Mage.Sets/src/mage/cards/b/BeaconOfCreation.java

```diff
@@ -1,37 +1,44 @@
 
 package mage.cards.b;
 
-import java.util.UUID;
+import mage.abilities.dynamicvalue.DynamicValue;
 import mage.abilities.dynamicvalue.common.PermanentsOnBattlefieldCount;
 import mage.abilities.effects.common.CreateTokenEffect;
 import mage.abilities.effects.common.ShuffleSpellEffect;
+import mage.abilities.hint.Hint;
+import mage.abilities.hint.ValueHint;
 import mage.cards.CardImpl;
 import mage.cards.CardSetInfo;
 import mage.constants.CardType;
 import mage.constants.SubType;
 import mage.filter.common.FilterControlledPermanent;
 import mage.game.permanent.token.InsectToken;
 
+import java.util.UUID;
+
 /**
- *
  * @author North
  */
 public final class BeaconOfCreation extends CardImpl {
 
+    private static final DynamicValue xValue = new PermanentsOnBattlefieldCount(new FilterControlledPermanent(SubType.FOREST));
+    private static final Hint hint = new ValueHint("Forests you control", xValue);
+
     private static final FilterControlledPermanent filter = new FilterControlledPermanent("Forest you control");
 
     static {
         filter.add(SubType.FOREST.getPredicate());
     }
 
     public BeaconOfCreation(UUID ownerId, CardSetInfo setInfo) {
-        super(ownerId,setInfo,new CardType[]{CardType.SORCERY},"{3}{G}");
+        super(ownerId, setInfo, new CardType[]{CardType.SORCERY}, "{3}{G}");
 
 
         // Create a 1/1 green Insect creature token for each Forest you control.
         this.getSpellAbility().addEffect(new CreateTokenEffect(new InsectToken(), new PermanentsOnBattlefieldCount(filter)));
```

Comment:
If possible then must use shared xValue in both hint and effect:

![shot_240904_155253](https://github.com/user-attachments/assets/05406d6e-7d57-470f-b121-baed5c753d68)